universite-du-quebec-a-montreal-prenoms.csl

```diff
@@ -849,55 +901,48 @@
           </group>
         </group>
       </else-if>
+      <else-if type="article-journal" match="any">
+        <text variable="genre" prefix="[" suffix="]"/>
+      </else-if>
       <else>
         <choose>
-          <if type="musical_score book article" match="none">
+          <if type="musical_score book article document" match="none">
             <group delimiter=". ">
               <choose>
                 <if type="manuscript" match="none">
                   <text variable="archive" prefix=". "/>
                 </if>
               </choose>
-              <group delimiter=" " prefix="[" suffix="]">
-                <text variable="medium"/>
+              <group delimiter="; " prefix="[" suffix="]">
                 <text variable="genre"/>
+                <text variable="medium"/>
               </group>
               <group delimiter=" : ">
                 <text term="scale" text-case="capitalize-first"/>
                 <text variable="scale"/>
               </group>
-              <number variable="number-of-pages"/>
             </group>
           </if>
+          <else-if type="document" match="any">
+            <text variable="genre"/>
+          </else-if>
         </choose>
       </else>
     </choose>
   </macro>
   <macro name="container-title-extra">
     <choose>
-      <if type="map" match="all">
-        <text variable="collection-title" prefix=". "/>
-        <text variable="number" prefix=", "/>
+      <if type="article-journal" match="all" variable="issued">
+        <text variable="status" prefix=". "/>
       </if>
-      <else-if type="article" match="any">
-        <text variable="number" prefix=" (" suffix=")"/>
-        <choose>
-          <if match="none" variable="author editor">
-            <text variable="genre" prefix=". [" suffix="]"/>
-          </if>
-          <else>
-            <text variable="genre" prefix=" [" suffix="]"/>
-          </else>
-        </choose>
-      </else-if>
     </choose>
   </macro>
-  <citation et-al-min="3" et-al-use-first="1" et-al-subsequent-min="3" et-al-subsequent-use-first="1" disambiguate-add-year-suffix="true" collapse="year">
+  <citation name-form="short" delimiter-precedes-last="never" et-al-min="3" et-al-use-first="1" et-al-subsequent-min="3" et-al-subsequent-use-first="1" disambiguate-add-year-suffix="true" collapse="year">
```

Comment:
no need to specify separate `et-al-subsequent` if it's the same as the general `et-al`